### ISSUE TITLE
[Research] Adding guidepup to storybook/demos automated tests

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3295]
### Provide a general summary of the issue here

Research if it's beneficial for our project to use [guidepup (screen reader driver for test automation)](https://www.guidepup.dev/docs/intro)

Will it allow us to find issues faster?

Does it need to be used in Storybook or Demos or both?

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS

**Comment by a11ymiko** [comment]
Using guidepup for Demos allows to automatically turn on NVDA/VoiceOver and check what is screen reader announcement for each component. 

**Pros:** 
- repeated scenarios can be tested without much effort
- real screen reader's output can be tested 

**Cons:**
- it helps find issues we already find with playwright (accessible name, accessible description, role)
- hard to setup and write tests that works as intended 
- not all interactions that can be tested manually can be tested with guidepup
- tests take a loooooong time

Using guidepup Virtual Screen Reader allows to get text of all element's properties from a11y tree in Interaction testing in Storybook. 

**Pros:**
- we can get better a11y test coverage on early stage
- fast
- can be easily used in CI

**Cons:**
- it involves tester to write changes in **.story.ts files
- it allows us to test things we already test with our unit tests
- it cannot find screen NVDA/VO/JAWS specific issues

**Comment by a11ymiko** [comment]
Decision is to only use Guidepup VSR in tests in Demos as its reliable at the moment. At the mean time I'll observe if there are any upgrades to Guidepup screen reader automation that could be beneficial for us.
